File: ai/typescript/detect-openai.ts
// ruleid: detect-openai
import OpenAI from "openai";

OPENAI_API_KEY = "asdf"

// ruleid: detect-openai
const openai = new OpenAI({
    apiKey: process.env.OPENAI_API_KEY,
});

// ruleid: detect-openai
const chatCompletion = await openai.chat.completions.create({
    messages: [{ role: "user", content: "Say this is a test" }],
    model: "gpt-3.5-turbo",
});

File: ai/python/detect-openai.py
OPENAI_API_KEY = "MY_API_KEY"
# ruleid: detect-openai
from openai import OpenAI
# ruleid: detect-openai
client = OpenAI(
    # Defaults to os.environ.get("OPENAI_API_KEY")
)
# ruleid: detect-openai
chat_completion = client.chat.completions.create(
    model="gpt-3.5-turbo",
    messages=[{"role": "user", "content": "Hello world"}]
)